File: onDeck/Utilities/NameCleaner.swift
import Foundation

nonisolated enum NameCleaner {
    /// Strips position suffixes (-P, -H, -DH) from Fantrax player names.
    /// Example: "Shohei Ohtani-P" -> "Shohei Ohtani"
    static func stripPositionSuffix(_ name: String) -> String {
        name.replacingOccurrences(of: #"-(P|H|DH)$"#, with: "", options: .regularExpression)
    }

    /// Strips periods from names for MLB API search compatibility.
    /// Example: "T.J. Rumfield" -> "TJ Rumfield"
    static func stripPeriods(_ name: String) -> String {
        name.replacingOccurrences(of: ".", with: "")
    }

    /// Full cleanup pipeline for Fantrax names before MLB API lookup.
    static func clean(_ name: String) -> String {
        stripPeriods(stripPositionSuffix(name))
    }
}

File: windows/src/OnDeck.Core/Utilities/NameCleaner.cs
using System.Text.RegularExpressions;

namespace OnDeck.Core.Utilities;

/// <summary>Port of <c>Utilities/NameCleaner.swift</c>.</summary>
public static partial class NameCleaner
{
    [GeneratedRegex(@"-(P|H|DH)$")]
    private static partial Regex PositionSuffixRegex();

    /// <summary>
    /// Strips position suffixes (-P, -H, -DH) from Fantrax player names.
    /// Example: "Shohei Ohtani-P" -> "Shohei Ohtani".
    /// </summary>
    public static string StripPositionSuffix(string name) =>
        PositionSuffixRegex().Replace(name, string.Empty);

    /// <summary>
    /// Strips periods from names for MLB API search compatibility.
    /// Example: "T.J. Rumfield" -> "TJ Rumfield".
    /// </summary>
    public static string StripPeriods(string name) => name.Replace(".", string.Empty);

    /// <summary>Full cleanup pipeline for Fantrax names before MLB API lookup.</summary>
    public static string Clean(string name) => StripPeriods(StripPositionSuffix(name));
}
